### ISSUE TITLE
Bootloop with latest 25.0 beta

### ISSUE THREAD

**Issue by 3arthur6** [topjohnwu/Magisk#5991]
Device: S20 Ultra (SM-G988N)
Android version: 12.0
Magisk version name: cacf8736
Magisk version code: 25001

My S20 isn't able to boot with latest beta release.
It just bootloop on first samsung logo before starting android.
It runs on my own custom kernel I attached with last_kmsg.

Could the kernel be the culprit?

Thanks in advance

[last_kmsg.txt](https://github.com/topjohnwu/Magisk/files/8921207/last_kmsg.txt)
[boot.zip](https://github.com/topjohnwu/Magisk/files/8921197/boot.zip)

**Comment by canyie** [comment]
```
<12>[    2.418187]  [1:           init:    1] magiskinit: open: /proc/bootconfig failed with 2: No such file or directory
<12>[    2.418488]  [1:           init:    1] magiskinit: Device config:
<12>[    2.418520]  [1:           init:    1] magiskinit: skip_initramfs=[0]
<12>[    2.418546]  [1:           init:    1] magiskinit: force_normal_boot=[0]
<12>[    2.418571]  [1:           init:    1] magiskinit: rootwait=[0]
<12>[    2.418595]  [1:           init:    1] magiskinit: slot=[]
<12>[    2.418619]  [1:           init:    1] magiskinit: dt_dir=[/proc/device-tree/firmware/android]
<12>[    2.418644]  [1:           init:    1] magiskinit: fstab_suffix=[default]
<12>[    2.418668]  [1:           init:    1] magiskinit: hardware=[qcom]
<12>[    2.418692]  [1:           init:    1] magiskinit: hardware.platform=[]
<12>[    2.418716]  [1:           init:    1] magiskinit: emulator=[0]
<12>[    2.419303]  [1:           init:    1] magiskinit: Found pattern [selinux_setup]
<12>[    2.419420]  [1:           init:    1] magiskinit: FirstStageInit
<12>[    2.424879]  [1:           init:    1] magiskinit: Replace [/system/bin/init] -> [/data/magiskinit]
<12>[    2.457269]  [1:           init:    1] magiskinit: Unmount [/sys]
<12>[    2.457345]  [1:           init:    1] magiskinit: Unmount [/proc]
<6>[    2.468931]  [6:           init:    1] max77705_get_charger_state : charger status (0x31)
<6>[    2.470276]  [6:           init:    1] mfc_s2miw04_chg_get_property cable_type =0
<4>[    2.486582]  [6:           init:    1] EXT4-fs (sda28): warning: maximal mount count reached, running e2fsck is recommended
<6>[    2.486982]  [6:           init:    1] EXT4-fs (sda28): mounted filesystem with ordered data mode. Opts: noauto_da_alloc,discard,journal_checksum,data=ordered,errors=panic
<0>[    2.491580]  [6:           init:    1] Kernel panic - not syncing: Attempted to kill init! exitcode=0x00007f00
<0>[    2.491580]  [6:           init:    1] 
<4>[    2.491589]  [6:           init:    1] CPU: 6 PID: 1 Comm: init FTT: 0 0 Tainted: G S                4.19.113/GVE4 #1
<4>[    2.491593]  [6:           init:    1] Hardware name: Samsung Z3Q KOR PROJECT - PV3 Rev05 (board-id,15) for HM1 EVT1 (DT)
```
does your device boot with v24.3?

**Comment by eclipsek20** [comment]
I can't start an issue because I don't meet the criteria so I'll just say it here, Zygisk is unable to activate on v25.0 beta on my phone (LG G8s) (Android 11), if someone has a similar phone please make an issue.

**Comment by 3arthur6** [comment]
> does your device boot with v24.3, or any version older than 24302?

Yep sure! It is rooted with it right now.
I can't find the debug version of 24.3 to get the last_kmsg from it.

**Comment by canyie** [comment]
[app-debug.zip](https://github.com/topjohnwu/Magisk/files/8933896/app-debug.zip)
can this build boot?

**Comment by 3arthur6** [comment]
> can this build boot?

Same issue.. Bootloop before android start

[last_kmsg.txt](https://github.com/topjohnwu/Magisk/files/8933950/last_kmsg.txt)

**Comment by canyie** [comment]
The last_kmsg does not contain magisk things and kernel panics so maybe you attached a wrong file 😵‍💫

**Comment by 3arthur6** [comment]
> The last_kmsg does not contain magisk things and kernel panics so maybe you attached a wrong file face_with_spiral_eyes Can you try it again and upload a fresh kmsg?

Whoops
Here it is. It looks the same..

[last_kmsg.txt](https://github.com/topjohnwu/Magisk/files/8935422/last_kmsg.txt)

**Comment by canyie** [comment]
Try this build [app-debug.zip](https://github.com/topjohnwu/Magisk/files/8933913/app-debug.zip)

**Comment by 3arthur6** [comment]
> Try this build [app-debug.zip](https://github.com/topjohnwu/Magisk/files/8933913/app-debug.zip)

Same again huh
[last_kmsg.txt](https://github.com/topjohnwu/Magisk/files/8935548/last_kmsg.txt)

**Comment by canyie** [comment]
Oh, maybe I uploaded wrong apk. Try this?
[app-debug.zip](https://github.com/topjohnwu/Magisk/files/8935575/app-debug.zip)

**Comment by 3arthur6** [comment]
> Oh, maybe I uploaded wrong apk. Try this? [app-debug.zip](https://github.com/topjohnwu/Magisk/files/8935575/app-debug.zip)

This time it stucks a Samsung logo. I have to force reboot it.
Last_kmsg is full of magiskinit loop stuff.

[last_kmsg.txt](https://github.com/topjohnwu/Magisk/files/8936033/last_kmsg.txt)

**Comment by 3arthur6** [comment]
I switch to the more general #6000 issue.

**Comment by canyie** [comment]
Oh, I made some typos that prevented it from working 😆 
[app-debug.zip](https://github.com/topjohnwu/Magisk/files/8936599/app-debug.zip)

> I switch to the more general https://github.com/topjohnwu/Magisk/issues/6000 issue.

No, it's not genernal, it's completely different issue! I think we should continue debugging in this issue.

**Comment by 3arthur6** [comment]
Ho sorry it looks very similar with #6000 
You right, it seems fixed for them but not for me.

Sadly your last built didn't fixed it..

[last_kmsg.txt](https://github.com/topjohnwu/Magisk/files/8936915/last_kmsg.txt)

**Comment by 3arthur6** [comment]
Ho sorry it looks very similar with #6000 
You right, it seems fixed for them but not for me.

Sadly your last built didn't fixed it..

[last_kmsg.txt](https://github.com/topjohnwu/Magisk/files/8936915/last_kmsg.txt)

**Comment by canyie** [comment]
I have removed all difference with v24.3 but it still bootloop. Could you take a kmsg with 24301?

**Comment by 3arthur6** [comment]
Thanks I finally found v24.3 in debug version by searching for 24301.
Here is the kmsg.
[kmsg.txt](https://github.com/topjohnwu/Magisk/files/8936992/kmsg.txt)

**Comment by canyie** [comment]
It seems your system's init not printing any logs, thus debugging becomes harder. 😢  Could you test if canary 24302 work?

**Comment by yujincheng08** [comment]
Duplicate of #5777
